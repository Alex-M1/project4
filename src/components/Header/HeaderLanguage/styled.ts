import styled from 'styled-components';
import { IStyled } from 'common/types/styledTypes';

export const StLangToggle = styled.button < IStyled > `
  position: absolute;
  top: 29px;
  right: 5%;
  background: transparent;
  color: #fff;
  border: none;
  font-size: 26px;
  cursor: pointer;
`;
