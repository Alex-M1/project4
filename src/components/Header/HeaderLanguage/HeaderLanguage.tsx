import React, { useState } from 'react';
import { useTranslation } from 'react-i18next';
import { useTheme } from 'src/components/hooks/useTheme';
import { StLangToggle } from './styled';

const HeaderLanguage: React.FC = () => {
  const { i18n } = useTranslation();
  const { colors, theme } = useTheme();
  const [lang, setLang] = useState<string>(localStorage.getItem('lang') || 'en');
  const translateHandler = () => {
    i18n.changeLanguage(lang === 'en' ? 'ru' : 'en');
    localStorage.setItem('lang', lang === 'en' ? 'ru' : 'en');
    setLang(lang === 'en' ? 'ru' : 'en');
  };
  return (
      <StLangToggle
          theme={theme}
          colors={colors}
          onClick={translateHandler}
      >
        {lang[0].toUpperCase() + lang.slice(1)}
      </StLangToggle>
  );
};
export default HeaderLanguage;