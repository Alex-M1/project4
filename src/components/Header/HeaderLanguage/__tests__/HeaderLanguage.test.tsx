import { mount, shallow } from 'enzyme';
import React from 'react';
import HeaderLanguage from '../HeaderLanguage';

jest.mock('react-i18next', () => ({
  useTranslation: jest.fn().mockImplementation(() => ({
    i18n: {
      changeLanguage: jest.fn(),
    },
  })),
}),
);

describe('HeaderLanguage', () => {
  it('Should match snapshot', () => {
    const component = shallow(<HeaderLanguage />);
    expect(component.html()).toMatchSnapshot();
  });
  it('should render StLangToggle', () => {
    const component = mount(<HeaderLanguage />);
    expect(component.find('styled__StLangToggle')).toHaveLength(1);
  });
  it('should render StLangToggle click & lang = ru', () => {
    const component = mount(<HeaderLanguage />);
    const langToggle = component.find('styled__StLangToggle');
    langToggle.simulate('click');
    expect(langToggle.getDOMNode().textContent).toBe('Ru');
  });

  it('should render StLangToggle click & lang = en', () => {
    localStorage.setItem('lang', 'ru');
    const component = mount(<HeaderLanguage />);
    const langToggle = component.find('styled__StLangToggle');
    langToggle.simulate('click');
    expect(langToggle.getDOMNode().textContent).toBe('En');
  });
});
