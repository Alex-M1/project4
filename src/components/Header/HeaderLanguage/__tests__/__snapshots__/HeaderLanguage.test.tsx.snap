// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderLanguage Should match snapshot 1`] = `"<button class=\\"styled__StLangToggle-gXeXkO cwoxAJ\\">En</button>"`;
